stdlib/src/builtin/tuple.mojo

```diff
@@ -178,3 +180,36 @@ struct Tuple[*element_types: Movable](Sized, Movable):
             The tuple element at the requested index.
         """
         return rebind[T](self[i])
+
+    @always_inline("nodebug")
+    fn __contains__[
+        inferred RHS_T: EqualityComparable
+    ](inout self, rhs: RHS_T) -> Bool:
+        """Verify if a given value is present in the tuple.
+
+        ```mojo
+        var x = Tuple(1,2,True)
+        if 1 in x: print("x contains 1")
+        ```
+        Args:
+            rhs: The value to find.
+
+        Parameters:
+            RHS_T: The inferred type of RHS. Must implement the
+              trait `EqualityComparable`.
+
+        Returns:
+            True if the value is contained in the tuple, False otherwise.
+        """
+        var result = False
+
+        @parameter
+        fn SingleIteration[Index: Int]():
+            if _type_is_eq[RHS_T, element_types[Index.value]]():
+                result |= rhs.__eq__(
+                    rebind[RHS_T](self.get[Index, element_types[Index.value]]())
```

Comment:
@laszlokindrat I don't think `rebind` will work for mem type.

Reply:
Good observation! I changed the implementation to not use rebind and changed rhs to owned instead of borrowed.

Tests runs fine on ubuntu locally but somehow fails on macosx here, and stop the testing process of ubuntu?

Please see results on the (macos-14)  tab: 
```mojo
Please file a bug report.
[6505:28327:20240517,222150.691559:WARNING crash_report_exception_handler.cc:257] UniversalExceptionRaise: (os/kern) failure (5)
```

Reply:
Why does the rhs need to be owned?

Reply:
In hope to not worry about borrowed references and manually extend lifetime with _=^,
Last commit passes tests, but there must be a better way to cast a reference than that ?

Reply:
> In hope to not worry about borrowed references and manually extend lifetime with _=^,

Unfortunately, `owned` can cause unintended copies, so I would rather avoid it.

Reply:
Hello, i forced-push a commit with changes related to both suggestions (no owned rhs and no rebind of T),
 i suggest rebinding the reference instead of T, since ```Reference``` is ```register_passable``` but not always T.
Extended test with memory-only types passes for ```Tuple.__contains__``` on last commit!
If both of you likes it, we could resolve this specific review conversation